optimum/exporters/ipex/model_patcher.py

```diff
@@ -94,6 +139,10 @@ def _patch_model(model):
         )
     if isinstance(model, LlamaForCausalLM):
         model = _patch_llama_model(model)
+    elif isinstance(model, FalconForCausalLM):
+        model = _patch_falcon_model(model)
+    elif isinstance(model, GPT2LMHeadModel):
+        model = _patch_gpt2_model(model)
```

Comment:
should this patch only applied for the text-generation task and if yes why ?

Reply:
Because we use IAKV in gpt2 and it only benefits when `use_cache=True`.